docs/content/direct/architecture.md

```diff
@@ -476,8 +476,10 @@ When processing a reference to a `Binding` object that still exists, the transpo
 
 Transport controller is based on the controller design patten and aims to bring the current state to the desired state. If a WEC was removed from the `Binding`, the transport controller will also make sure to remove the matching wrapped object(s) from the WEC's mailbox namespace.
 
-The transport controller maintains a cached set of customization properties for each destination, and an association between `Binding` and the set of destinations that it references. When a relevant informer delivers an event that indicates a change in a destination's properties, the controller invalidates that destination's entry in the cache and enqueues a reference to every relevant `Binding`. The template expansion work described above has a side-effect of putting entries in the destination properties cache.
\ No newline at end of file
+To support efficient application of the `CustomTransform` objects, the transport controller maintains a cache that maps a `metav1.GroupResource` to the result of digesting all the relevant `CustomTransform` objects. The data type `customTransformCollection` holds the cache. This cache is populated as needed in the course of transforming workload objects. To support eventually removing an entry from that cache when no longer needed, each entry also tracks the set of `Binding` objects that depend on that cache entry; when that set becomes empty, the entry is removed from the cache. In order to know when a `Binding` has changed from dependent to not dependent, the cache also tracks the set of `metav1.GroupResource` relevant to each `Binding`. Also: in order to know what to update when the Spec of a `CustomTransform` changes, the cache also tracks the spec of each `CustomTransform` that contributed to an entry in the cache. When such a `CustomTransformSpec` changes in any way, the controller invalidates the cache entry for the `GroupResource` of the last known version of that spec. Additionally, if the change affected the `GroupResource` then the controller also invlidates the cache entry for the old `GroupResource`.
```

Comment:
While this paragraph is technically correct, I think it can greatly benefit from a diagram that shows the flow visually.

Reply:
I have revised to add, hopefully, more better explanation.

Rather than trying to diagram the logic in each function, I started with an explanation of the data structure that they are maintaining. I gave a relational exposition, since it is simple and that is a widely known conceptual framework. With the relations and invariants shown, and the way that the data structure represents the relations shown, the explanations of the public methods of the cache can lean on that other material to be shorter and more comprehensible (I hope).